builders/interoperability/xcm/xc-registration/assets.md

```diff
@@ -39,193 +39,107 @@ To create a forum post on the [Moonbeam Community Forum](https://forum.moonbeam.
-        ```
-
-    === "Moonriver"
-
-        ```text
-        800000000
-        ```
-
-    === "Moonbase Alpha"
-
-        ```text
-        638978000
-        ```
-
-- `--target` or `--t` - (optional) target price for XCM execution, defaults to `$0.02`
-- `--asset` or `--a` - (optional) the token [Coingecko API ID](https://www.coingecko.com){target=\_blank}
-- `--price` or `--p` - (optional) if the Coingecko API does not support the token, you can specify the price manually
-
-For example, to calculate the units per second of DOT (Polkadot token), which has 10 decimals, on Moonbeam, you can run:
-
-```bash
-yarn calculate-units-per-second --d 10 --a polkadot --xwc 800000000 
-```
-
-Which should result in the following output (at the time of writing):  
-
-```text
-Token Price is $7.33
-The UnitsPerSecond needs to be set 34106412005
-```
-
 ### Generate the Encoded Calldata for the Asset Registration {: #generate-encoded-calldata-for-asset-registration }
 
 If you're not familiar with the governance system on Moonbeam, you can find out more information on the [Governance on Moonbeam](/learn/features/governance/){target=\_blank} page. With any governance proposal on Moonbeam, you'll need to submit a preimage, which defines the actions to be executed, and then use the preimage to submit a proposal.
 
-To submit a preimage, you'll need to get the encoded calldata for each extrinsic that you want to execute. As previously mentioned, you'll use the `assetManager.registerForeignAsset`, and optionally, the `assetManager.setAssetUnitsPerSecond`, and `system.setStorage` extrinsics.
+To submit a preimage, you'll need to get the encoded calldata for each extrinsic that you want to execute. As previously mentioned, you'll use the `evmForeignAssets.createForeignAsset` and the `xcmWeightTrader.addAsset` extrinsics.
 
-You can use the [`xcm-asset-registrator.ts` script](https://github.com/Moonsong-Labs/xcm-tools/blob/main/scripts/xcm-asset-registrator.ts){target=\_blank} to calculate the encoded calldata and even submit the preimage and proposal if you desire. Proposals must be submitted via the General Admin Track. If you're registering an asset and opening a channel, you'll want to wait to submit the preimage and proposal until you have the calldata for the channel-related calls.
+Proposals must be submitted via the `FastGeneralAdmin` track. If you're opening a channel and registering an asset and you'll want to wait until the channel is established prior to attempting to register the asset. 
 
-To get the encoded calldata for the `assetManager.registerForeignAsset` extrinsic, you can use the following arguments:
+To get the encoded calldata for the `evmForeignAssets.createForeignAsset` extrinsic, you will need to provide the following arguments:
 
-- `--ws-provider` or `--w` - the WebSocket provider to be used for the requests. The WSS network endpoints for each Moonbeam-based network are as follows:
+- `assetId` - unique identifier of the asset
```

Comment:
We never explain how to get the assetID, actually, because the user hast o provide the multilocation we can modify `xcm-tools` to create this for you